apps/server/src/modules/board/service/board-do-copy-service/swap-internal-links.visitor.ts

```diff
@@ -0,0 +1,60 @@
+import {
+	AnyBoardDo,
+	BoardCompositeVisitor,
+	Card,
+	Column,
+	ColumnBoard,
+	LinkElement,
+	SubmissionContainerElement,
+	SubmissionItem,
+} from '@shared/domain/domainobject';
+import { DrawingElement } from '@shared/domain/domainobject/board/drawing-element.do';
+import { EntityId } from '@shared/domain/types';
+
+export class SwapInternalLinksVisitor implements BoardCompositeVisitor {
+	constructor(map: Map<EntityId, EntityId>) {
+		this.idMap = map;
+	}
+
+	private idMap: Map<EntityId, EntityId>;
+
+	visitDrawingElement(drawingElement: DrawingElement): void {
+		this.visitChildrenOf(drawingElement);
+	}
+
+	visitCard(card: Card): void {
+		this.visitChildrenOf(card);
+	}
+
+	visitColumn(column: Column): void {
+		this.visitChildrenOf(column);
+	}
+
+	visitColumnBoard(columnBoard: ColumnBoard): void {
+		this.visitChildrenOf(columnBoard);
+	}
+
+	visitExternalToolElement(): void {}
```

Comment:
Are these just no-ops or is it an error when they get called?

Reply:
its valid to visit them, just nothing to do